internal/controller/nnf_storage_controller.go

```diff
@@ -915,6 +938,96 @@ func (r *NnfStorageReconciler) getFsName(ctx context.Context, nnfStorage *nnfv1a
 
 }
 
+func (r *NnfStorageReconciler) runSharedMGTCommands(ctx context.Context, nnfStorage *nnfv1alpha6.NnfStorage) (*ctrl.Result, error) {
+	nnfStorageProfile, err := getPinnedStorageProfileFromLabel(ctx, r.Client, nnfStorage)
+	if err != nil {
+		return nil, dwsv1alpha3.NewResourceError("could not find pinned storage profile").WithError(err).WithFatal()
+	}
+
+	// Some tests don't fake out the nnfStorage completely
+	if len(nnfStorage.Spec.AllocationSets) == 0 {
+		return nil, nil
+	}
+
+	// If this NnfStorage is for a standalone MGT, then we don't need to run the shared MGT commands
+	if len(nnfStorage.Spec.AllocationSets) == 1 && nnfStorage.Spec.AllocationSets[0].Name == "mgt" {
+		return nil, nil
+	}
+
+	nnfLustreMgt, err := r.getLustreMgt(ctx, nnfStorage)
+	if err != nil {
+		return nil, dwsv1alpha3.NewResourceError("could not get NnfLustreMGT for address: %s", nnfStorage.Status.MgsAddress).WithError(err)
+	}
+
+	// An NnfLustreMgt in the "nnf-system" namespace indicates an external MGT. Commands can't be run on external MGTs.
+	if nnfLustreMgt.GetNamespace() == "nnf-system" {
+		return nil, nil
```

Comment:
Same here maybe? (re: log)

Reply:
I can add a comment to this one in case anyone gets confused why their MGT commands aren't working for an external MGT. I think we're the only ones that run that way, though.